doc/Programming-guide/Using-Kaa-endpoint-SDKs/C/SDK-Linux/index.md

```diff
@@ -2,304 +2,198 @@
+        {
+            for (size_t i=1; i<KAA_APP_PRINT_TIMES; i++) {
+                printf("Hello, I am a Kaa Application!\n");
+            }
+            kaa_client_stop(context);
+        }
 
-   ~~~
-   $ wget https://github.com/randombit/botan/archive/1.11.28.tar.gz
-   $ tar -zxf 1.11.28.tar.gz
-   $ cd botan-1.11.28/
-   $ ./configure.py
-   $ sudo make install
-   $ sudo ln -s /usr/local/include/botan-1.11/botan /usr/local/include/botan
-   ~~~
+        int main()
+        {
+            kaa_client_t* kaa_client = NULL;
+            kaa_error_t error = kaa_client_create(&kaa_client, NULL);
+            if (error) {
+                return -1;
+            }
 
-6. Install the **[SQLite](https://www.sqlite.org/index.html)** library by executing the following command:
-    1. For automatic installation, execute the following commands (tested on Ubuntu 14.04):
+            error = kaa_client_start(kaa_client, &dummy_function, (void *)kaa_client, 0);
 
-       ~~~
-       $ sudo apt-get install libsqlite3-0 libsqlite3-dev
-       ~~~
+            kaa_client_destroy(kaa_client);
 
-    2. For manual installation, refer to the following example:
+            if (error) {
+                return -1;
+            }
 
-       ~~~
-       $ wget https://www.sqlite.org/2015/sqlite-autoconf-3081002.tar.gz
-       $ tar -zxf sqlite-autoconf-3081002.tar.gz
-       $ cd sqlite-autoconf-3081002/
-       $ ./configure
-       $ sudo make install
-       ~~~
+            return 0;
+        }
 
-**NOTE:** Instead of manually installing all required components and libraries, you can follow [the quick way to build C/C++ endpoint SDK](#Linux-Quickway).  (only applicable for x86\_64 platform build)
+    The application code is simple: application just prints *KAA_APP_PRINT_TIMES* times the string *"Hello, I am a Kaa Application!\n"*. That's it. So, now we have the following file structure:
 
-## Compile C++ endpoint SDK
+        - my-kaa-application
+           - kaa
+             - Unpacked C SDK
+           - src
+             - kaa-application.c
 
-To build the C++ endpoint SDK, do the following:
+5. Create CMakeLists.txt file in the my-kaa-application directory. It is a top-level cmake file which is responsible for the application build.
 
-<!-- TODO: Fix this link when Administration-UI-guide#AdministrationUIguide-GeneratingSDK are updated -->
-1. [Generate](Administration-UI-guide#AdministrationUIguide-GeneratingSDK) the C++ endpoint SDK in Admin UI.
-2. Download and untar the Kaa C++ SDK archive.
+        touch CMakeLists.txt
 
-   ~~~ bash
-   $ tar xfv kaa-cpp-ep-sdk.tar.gz
-   ~~~
+6. Add the following code to top-level CMakeLists.txt:
 
-3. Run the following commands.
+        cmake_minimum_required(VERSION 2.8.8)
+        project(kaa-application C)
 
-   ~~~
-   mkdir build
-   cd build
-   cmake ..
-   make
-   make install
-   ~~~
+        find_package(OpenSSL REQUIRED)
 
-## Configure C++ endpoint SDK
+        set(APP_NAME "kaa-app")
```

Comment:
For this example, I think there is no need to reference `kaa-app` using variable. It's up to you though.